src/sentry/api/event_search.py

```diff
@@ -926,11 +930,16 @@ def convert_search_filter_to_snuba_query(search_filter, key=None, params=None):
     elif name in ARRAY_FIELDS and search_filter.value.raw_value == "":
         return [["notEmpty", [name]], "=", 1 if search_filter.operator == "!=" else 0]
     else:
-        value = (
-            int(to_timestamp(value)) * 1000
-            if isinstance(value, datetime) and name != "timestamp"
-            else value
-        )
+        # timestamp{,.to_{hour,day}} need a datetime string
+        # last_seen needs an integer
+        if isinstance(value, datetime) and not name.startswith("timestamp"):
```

Comment:
Can we be explicit about which `name`s we do this transformation on?

Reply:
uhhh so i previously asked @wedamija about why he put this condition in and he said:

> i have no idea, sorry. that was a while ago
> i assume this is just a value that clickhouse wants for dates

so i deleted it, and it caused a bunch of tests to fail, including `last_seen`, so i backed it out. i could go through the tests and make them work but i'm not confident that the tests cover all the cases.

Reply:
🤔 I'm curious why we're not confident tests cover all the cases, there should be a pretty limited set of fields that are time related.

> including `last_seen`

Could we update the comment to include all the fields we're aware of that depend on this at least?

Reply:
well... it was partly "i'm not confident that the tests are 100% coverage" and partly "this seems very complicated to track down and the hackaround is only three lines" and also we might be getting rid of this or significantly changing it for snql anyways?

Reply:
I think you can change this to do something like
```python
name not in {"timestamp", "timestamp.to_hour", "timestamp.to_day"}
```
(probably generate this set instead of hard coding) with little to no concerns for breaking anything else. And that way we're not just matching starts with `timestamp` and risk breaking something else.

Reply:
what do you mean by "probably generate this set instead of hard coding"?

Reply:
Not saying this is exactly what you should do but something like checking the `FIELD_ALIASES` for `timestamp.to_*` fields and generate a set from that so if new field aliases get added, they're handled here without having to manually add it.

Reply:
@Zylphrex's suggestion implemented in https://github.com/getsentry/sentry/pull/24530/commits/dad1e13ff94215913a55a64f116abd928da19542. I didn't thoroughly investigate why tests fail due to https://github.com/getsentry/sentry/pull/24530#discussion_r597813105, but recorded them in https://gist.github.com/alxu-sentry/4105998bcec59fc2812466ce2bb4d8b3. Thank you for the feedback.